internal/run/run.go

```diff
@@ -774,7 +774,25 @@ func (w *Workflow) snapshotSource(ctx context.Context, parentStep *workflowTrack
 }
 
 func (w *Workflow) getRegistryTags(ctx context.Context, sourceID string) ([]string, error) {
-	tags := []string{"latest"}
+	var tags []string
+	if env.IsGithubAction() {
+		tags = append(tags, "latest")
+
+		// implicitly add branch tag
+		var branch string
+		if strings.Contains(os.Getenv("GITHUB_REF"), "refs/heads/") {
+			branch = strings.TrimPrefix(os.Getenv("GITHUB_REF"), "refs/heads/")
+		} else if strings.Contains(os.Getenv("GITHUB_REF"), "refs/pull/") {
+			branch = strings.TrimPrefix(os.Getenv("GITHUB_HEAD_REF"), "refs/heads/")
+		}
+
+		// trim to fit docker tag format
+		branch = strings.Trim(branch, " ")
```

Comment:
nit: thoughts on using the inbuilt method?

```suggestion
		branch = strings.TrimSpace(branch)
```

similarly for replace -1, why not

```
		branch = strings.ReplaceAll(branch, "/", "-")
```

Reply:
Got some time on your hands? :upside_down_face: